### ISSUE TITLE
 Cannot find the DispatchAdapter's namespace for using CoreWebView2.AddHostObjectToScript with WinUI3

### ISSUE THREAD

**Issue by Take-A-Byte** [MicrosoftEdge/WebView2Feedback#2937]
**Description**

When following [the guide to add a native object to JS](https://learn.microsoft.com/en-us/microsoft-edge/webview2/how-to/winrt-from-js?tabs=csharp) in the WinUI3 project, the compiler complains that it is not able to find the `WinRTAdapter` namespace i.e. DispatchAdapter's namespace

**Version**
SDK: (one included with WinAppSDK 1.1.3)
Framework: WinAppSDK/WinUI3
OS: Win11

**Repro Steps**

In https://learn.microsoft.com/en-us/microsoft-edge/webview2/how-to/winrt-from-js?tabs=csharp, instead of **webview2_sample_uwp** project I have a WinUI3 project - say **webview2_sample_winui3**

1. Add CsWinRT nuget package to the **webview2_sample_winui3** project
2. Follow the first 6 steps in the above guide.
3. When we try to add the code in step 7 i.e.
```
private async void InitializeWebView2Async()
{
   await WebView2.EnsureCoreWebView2Async();
   var dispatchAdapter = new WinRTAdapter.DispatchAdapter();
   WebView2.CoreWebView2.AddHostObjectToScript("Windows", dispatchAdapter.WrapNamedObject("Windows", dispatchAdapter));
}
``` 
the compiler complains that the `WinRTAdapter` namespace does not exist.

When doing the same with the UWP project, it is able to find the namespace **successfully**, I expected the same with WinUI3.

**Comment by Take-A-Byte** [comment]
I just found out the solution that [krschau](https://github.com/krschau) provided for this. Here is the[ link for the comment](https://github.com/microsoft/microsoft-ui-xaml/issues/7170#issuecomment-1215910370).

@Diegorro98 does this solve your problem as well? If yes, I would like to close the issue.

**Comment by Diegorro98** [comment]
Well... It works, but I can't do anything with this because I need to create objects and call methods from the main project.
But the problem is solved, so feel free to close the issue.

**Comment by nsandeep87** [comment]
@Take-A-Byte  and @Diegorro98 : Can anybody please guide me here. I took the project mentioned by @krschau
https://github.com/krschau/WinUI-WebView2-Samples/tree/master/NativeWebWinUI3App

This project builds  properly for me. If I right click the WinRTAdapter properties mentioned in the above project, I don't see any Webview2 options under Common properties( basically to configure include filters). Also from my Javascript, if I run **_chrome.webview.hostobjects.sync.CppWinrtRuntimeComponent_** I get an error.
Please let me know if I am missing out anything here.

**Comment by Take-A-Byte** [comment]
@nsandeep87 use small "a" in "adjustTemperature". When injecting a WinRT object like this into a JavasScript namespace all method names will have their first letter lowercased.

**Comment by nsandeep87** [comment]
Thanks @Take-A-Byte for correcting me. Works fine with adjustTemperature.

**Comment by nsandeep87** [comment]
@Take-A-Byte  @Diegorro98 @david-risney 
I have run into another issue here. As discussed above, the projection of CppWIndowsRuntime works fine as expected. However I run into issues with C# WindowsRuntime. We cannot add project refs of C# Windows Runtime on the WinUI3 project. As a result of this when we try to instantiate the class on the JS as shown below, I get ther error "Class not registered"
```
var csruntime = chrome.webview.hostObjects.sync.CSharpRuntime // CSharpRuntime is my windows runtime project written in c#
var runtimeInstance= new csruntime.Class1() // I get the error Class not found
```
Does this work with C# Windows runtime. Please let me know if anybody has tried it

**Comment by david-risney** [comment]
"Class not registered" for WinRT runtimeclasses usually means registration information (like mapping from the class metadata to the implementation DLL) is missing for that class. Are you able to use the WinRT class from a project that class isn't defined in, and from C++ or C# not via wv2winrt? If that doesn't work then it may be an issue with the registration of the runtimeclass.

Although in the comment you say "Class not found" which might not be related to WinRT registration. Can you paste the exact error message?

**Comment by nsandeep87** [comment]
@david-risney : The error that I am seeing is Class not registered
I tried to access directly( and not via wv2winrt) and these are my observations. Let me know if I need to try anything else

Add reference by add project
1. Create a WinRT c# project.
2. Add the project reference to c++/c# WinUI3
3. But the WinUI3 fails to compile saying the **_WinRT project is not compatible with net6.0-windows10.0.2200. WinRT project supports only uap10.0.2200_**

I also tried to add reference via Assembly
1. Create WinRT project. Build and generate a winmd
2. Added the winmd reference a c++/c# WinUI3 project.
3. But I could not import the namespace as it was not found

**Comment by Take-A-Byte** [comment]
@nsandeep87 In my opinion it is always better to create a new issue if there is some unsolved problem. This was all the developers would be able to see the issue and provide you with a solution after checking the `Open` issue. IMHO, developers might miss out on `Closed` issues, thinking that the problem has been solved and even that by default Github would only show `Open` issues.

By the way, if the constructor of `Class1` refers to something static, it did not work for me and I also had to include all the namespaces of types that the constructor uses. I hope this helps 🤞
